src/index.js

```diff
@@ -0,0 +1,227 @@
+      }, this);
+    }
+  }
+
+  renderExtractedChunk(chunk) {
+    const source = new ConcatSource();
+    chunk.forEachModule((module) => {
+      const moduleSource = module.source();
+      source.add(this.applyAdditionalInformation(moduleSource, module.additionalInformation));
+    }, this);
+    return source;
+  }
+
+  extract(options) {
+    if (Array.isArray(options) || isString(options) || typeof options.options === 'object' || typeof options.query === 'object') {
+      options = { use: options };
+    } else {
+      validateOptions(path.resolve(__dirname, '../schema/loader.json'), options, 'Extract Text Plugin (Loader)');
+    }
+    let loader = options.use;
+    let before = options.fallback || [];
+    if (isString(loader)) {
+      loader = loader.split('!');
+    }
+    if (isString(before)) {
+      before = before.split('!');
+    } else if (!Array.isArray(before)) {
+      before = [before];
+    }
+    options = mergeOptions({ omit: before.length, remove: true }, options);
+    delete options.use;
+    delete options.fallback;
+    return [this.loader(options)]
+      .concat(before, loader)
+      .map(getLoaderObject);
+  }
+
+  apply(compiler) {
+    const options = this.options;
+    compiler.plugin('this-compilation', (compilation) => {
+      const extractCompilation = new ExtractTextPluginCompilation();
+      compilation.plugin('normal-module-loader', (loaderContext, module) => {
+        loaderContext[NS] = (content, opt) => {
+          if (options.disable) { return false; }
+          if (!Array.isArray(content) && content != null) { throw new Error(`Exported value was not extracted as an array: ${JSON.stringify(content)}`); }
+          module[NS] = {
+            content,
+            options: opt || {},
+          };
+          return options.allChunks || module[`${NS}/extract`]; // eslint-disable-line no-path-concat
+        };
+      });
+      const filename = this.filename;
+      const id = this.id;
+      let extractedChunks;
+      compilation.plugin('optimize-tree', (chunks, modules, callback) => {
+        extractedChunks = chunks.map(() => new Chunk());
+        chunks.forEach((chunk, i) => {
+          const extractedChunk = extractedChunks[i];
+          extractedChunk.index = i;
+          extractedChunk.originalChunk = chunk;
+          extractedChunk.name = chunk.name;
+          extractedChunk.entrypoints = chunk.entrypoints;
+          chunk.chunks.forEach((c) => {
+            extractedChunk.addChunk(extractedChunks[chunks.indexOf(c)]);
+          });
+          chunk.parents.forEach((c) => {
+            extractedChunk.addParent(extractedChunks[chunks.indexOf(c)]);
+          });
+        });
+        async.forEach(chunks, (chunk, callback) => { // eslint-disable-line no-shadow
+          const extractedChunk = extractedChunks[chunks.indexOf(chunk)];
+          const shouldExtract = !!(options.allChunks || isInitialOrHasNoParents(chunk));
+          chunk.sortModules(module);
```

Comment:
I'm seeing https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/565 in a webpack project at $WORK. I don't think you should be passing in `module` to `sortModules` (since it isn't a comparator function) and making this change fixes the aforementioned problem for me.